lib/driver.js

```diff
@@ -292,13 +292,9 @@ class XCUITestDriver extends BaseDriver {
     this.logEvent('xcodeDetailsRetrieved');
 
     // at this point if there is no platformVersion, get it from the device
-    if (!this.opts.platformVersion) {
-      if (this.opts.device && _.isFunction(this.opts.device.getPlatformVersion)) {
-        this.opts.platformVersion = await this.opts.device.getPlatformVersion();
-        log.info(`No platformVersion specified. Using device version: '${this.opts.platformVersion}'`);
-      } else {
-        // TODO: this is when it is a real device. when we have a real object wire it in
-      }
+    if (!this.opts.platformVersion && this.opts.device) {
+      this.opts.platformVersion = await this.opts.device.getPlatformVersion();
```

Comment:
No `getPlatformVersion()`?
https://github.com/appium/appium-ios-device/search?q=getPlatformVersion&unscoped_q=getPlatformVersion
(appium-ios-simulator for simulator has tho)

Reply:
The `this.opts.device` object is not an instance of `appium-ios-device`. I'm not sure what you're asking.

Reply:
Ah, I thought where the function was for real device since I could not find it in appium org.
But I understood `this.opts.platformVersion` was set in `determineDevice` before here, so `getPlatformVersion()` won't be called.